server/src/logic/handler/routes/recipe.handler.ts

```diff
@@ -115,4 +120,20 @@ export class RecipeHandler {
       .then(() => res.status(200).json())
       .catch((e: any) => res.status(e.statusCode).json(e.forResponse()));
   }
+
+  public uploadImage = (req: SocoboRequest, res: Response) => {
```

Comment:
i think this is looks really complicated. what did you think about this?

```ts
  public uploadImageV2 = (req: SocoboRequest, res: Response) => {
    const userEmail = req.requestData.decoded.email;
    const recipeId = req.params.id;

    this._imgService.persistImage(req.file.fieldname, DataType.RECIPE_IMAGE, userEmail)
      .then((url: string) => this._getRecipe(recipeId, url))
      .then((obj: any) => this._addImage(obj.recipe, obj.url))
      .then((recipe: Recipe) => this._db.recipes.update(recipeId, recipe))
      .then(() => this.getById(req, res))
      .catch((e: any) => res.status(e.statusCode).json(e.forResponse()));
  }

  private _getRecipe = (recipeId: number, url: string): Promise<any> => {
    return this._db.recipes.getById(recipeId).then(recipe => Promise.resolve({recipe, url}));
  }

  private _addImage = (recipe: Recipe, url: string): Promise<any> => {
    recipe.addImageUrl(url);
    return Promise.resolve(recipe);
  }
```

it's not tested, because i don't know to send an image POST request.